python/pyarrow/tests/test_convert_builtin.py

```diff
@@ -297,10 +298,23 @@ def test_sequence_numpy_integer_inferred(seq, np_scalar_pa_type):
     arr = pa.array(seq(expected))
```

Comment:
Is the special case in the `else` block above still required?

Reply:
Yes. There's an xfail below for converting uint64 maximum